### PR TITLE
Add info about soft and hard deletes

### PR DESCRIPTION
**Pull request by lzdanski**
- https://github.com/astronomer/docs/issues/2244
- Included steps for enabling and completing a hard delete.
- Added tip about regular audits to confirm deletion of databases.
- Added definitions for hard delete vs soft delete.

### CODE REVIEW

File: software/configure-deployment.md [hunk 1]
```diff
@@ -187,22 +187,32 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
 ## Delete a Deployment
 
-You can delete an Airflow Deployment using the **Delete Deployment** button at the bottom of the Deployment's **Settings** tab.
+You can delete an Airflow Deployment using the **Delete Deployment** button at the bottom of the Deployment's **Settings** tab. 
 
-When you delete a Deployment, your Airflow webserver, scheduler, metadata database, and deploy history will be deleted, and you will lose any configurations set in the Airflow UI.
+When you delete a Deployment, you delete your Airflow webserver, scheduler, metadata database, and deploy history, and you lose any configurations set in the Airflow UI. 
 
-In your Astronomer database, the corresponding `Deployment` record will be given a `deletedAt` value and continue to persist until permanently deleted.
+By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database.
```
[the remark this answers is not in the document]
Comment (rishkarajgi) [reply software/configure-deployment.md:194]: Additionally the cleanup job can be tweaked via the astronomer helm chart

`astronomer.houston.cleanupDeployments`

```
# Cleanup deployments that have been soft-deleted in Houston
  # This runs as a CronJob
  cleanupDeployments:
    # Enable cleanup CronJob
    enabled: true

    # Default here is to run at midnight every night https://crontab.guru/#0_0_*_*_*
    schedule: "0 0 * * *"

    # Cleanup deployments older than this many days
    olderThan: 14
```

**Review by rishkarajgi — APPROVED** [review]

File: software/configure-deployment.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rishkarajgi) [reply software/configure-deployment.md]: lgtm

File: software/configure-deployment.md [hunk 2]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
+You can delete an Airflow Deployment using the **Delete Deployment** button at the bottom of the Deployment's **Settings** tab. 
 
-When you delete a Deployment, your Airflow webserver, scheduler, metadata database, and deploy history will be deleted, and you will lose any configurations set in the Airflow UI.
+When you delete a Deployment, you delete your Airflow webserver, scheduler, metadata database, and deploy history, and you lose any configurations set in the Airflow UI. 
 
-In your Astronomer database, the corresponding `Deployment` record will be given a `deletedAt` value and continue to persist until permanently deleted.
+By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any persisting values from your soft delete.
```
Comment (jwitz) [inline software/configure-deployment.md:194]:
```suggestion
By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any values that remained after your soft delete.
```

Comment (jwitz) [inline software/configure-deployment.md:194]:
```suggestion
By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any values that remained after your soft delete.
```

File: software/configure-deployment.md [hunk 3]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
+
+:::
+
+### Automate Deployment deletion clean up 
+
+Astronomer runs a cronjob to hard delete the deleted Deployment's metadata database and Deployment entry in your Astronomer database at midnight on a specified day. You can enable whether this cronjob runs or not, how many days after your soft delete to run the cronjob, and what time of day to run the cronjob by editing `astronomer.houston.cleanupDeployments` in your Astronomer Helm chart.
+
+The following example shows the default settings you can find in your Helm chart.
+
+```
```
Comment (jwitz) [inline software/configure-deployment.md:208]:
```suggestion
```yaml
```

File: software/configure-deployment.md [hunk 4]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
+
+Astronomer recommends regularly doing a database audit to confirm that you hard delete databases.
+
+:::
+
+### Automate Deployment deletion clean up 
+
+Astronomer runs a cronjob to hard delete the deleted Deployment's metadata database and Deployment entry in your Astronomer database at midnight on a specified day. You can enable whether this cronjob runs or not, how many days after your soft delete to run the cronjob, and what time of day to run the cronjob by editing `astronomer.houston.cleanupDeployments` in your Astronomer Helm chart.
+
+The following example shows the default settings you can find in your Helm chart.
```
Comment (jwitz) [inline software/configure-deployment.md:206]:
```suggestion
The following is an example of how you might configure the cronjob in your Helm chart: 
```

Comment (jwitz) [inline software/configure-deployment.md:206]:
```suggestion
The following is an example of how you might configure the cronjob in your Helm chart: 
```

File: software/configure-deployment.md [hunk 5]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
+To reuse a custom release name given to an existing Deployment after a soft delete but before Astronomer automatically cleans up any persisting Deployment records, you need to hard delete both the Deployment's metadata database and the Deployment's entry in your Astronomer database. 
 
-To reuse a custom release name given to an existing Deployment, you need to first hard delete both the Deployment's metadata database and the Deployment's entry in your Astronomer database. To do so, select the **Hard Delete?** checkbox before clicking **Delete Deployment**. Alternatively, you can run `astro deployment delete --hard` via the Astro CLI.
+1. Enable hard delete as an option at the platform level. To enable this feature, set `astronomer.houston.config.deployments.hardDeleteDeployment: true` in your `config.yaml` file and push the changes to your platform as described in [Apply a config change](apply-platform-config.md).
 
-![Hard delete checkbox](/img/software/hard-delete.png)
+2. You can delete a Deployment with the UI or CLI.
```
Comment (jwitz) [inline software/configure-deployment.md:228]:
```suggestion
2. Hard delete a Deployment with the Software UI or Astro CLI.
```

File: software/configure-deployment.md [hunk 6]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
+  - **In the UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**.
```
Comment (jwitz) [inline software/configure-deployment.md:229]:
```suggestion
  - **In the Software UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
```

Comment (jwitz) [inline software/configure-deployment.md:229]:
```suggestion
  - **Software UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
```

Comment (jwitz) [inline software/configure-deployment.md:229]:
```suggestion
  - **Software UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
  
    ![Hard delete checkbox](/img/software/hard-delete.png)
    
```

File: software/configure-deployment.md [hunk 7]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
 
-To reuse a custom release name given to an existing Deployment, you need to first hard delete both the Deployment's metadata database and the Deployment's entry in your Astronomer database. To do so, select the **Hard Delete?** checkbox before clicking **Delete Deployment**. Alternatively, you can run `astro deployment delete --hard` via the Astro CLI.
+1. Enable hard delete as an option at the platform level. To enable this feature, set `astronomer.houston.config.deployments.hardDeleteDeployment: true` in your `config.yaml` file and push the changes to your platform as described in [Apply a config change](apply-platform-config.md).
 
-![Hard delete checkbox](/img/software/hard-delete.png)
+2. You can delete a Deployment with the UI or CLI.
+  - **In the UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
+  - **Astro CLI:** Alternatively, you can run `astro deployment delete --hard`.
```
Comment (jwitz) [inline software/configure-deployment.md:230]:
```suggestion
  - **Astro CLI:** Run `astro deployment delete --hard`.
```

File: software/configure-deployment.md [hunk 8]
```diff
@@ -187,22 +187,52 @@ To deploy DAGs via git-sync, you add DAGs to a repository that has been configur
...
 
 This action permanently deletes all data associated with a Deployment, including the database and underlying Kubernetes resources.
 
+![Hard delete checkbox](/img/software/hard-delete.png)
```
Comment (jwitz) [inline software/configure-deployment.md:234]:
```suggestion
```

**Review by jwitz — APPROVED** [review]
LGTM! Only minor comments

File: software/configure-deployment.md (hunk 4 above)
Comment (lzdanski) [reply software/configure-deployment.md:206]: ah thank you for this. I was struggling with the phrasing here.

File: software/configure-deployment.md (hunk 6 above)
Comment (lzdanski) [reply software/configure-deployment.md:229]: added the most recent suggestion only

Comment (lzdanski) [reply software/configure-deployment.md:229]: added the most recent suggestion only

File: software/configure-deployment.md (hunk 4 above)
Comment (lzdanski) [reply software/configure-deployment.md:206]: added the most recent suggestion only

File: software/configure-deployment.md (hunk 2 above)
Comment (lzdanski) [reply software/configure-deployment.md:194]: added the most recent suggestion only

**Pull request merged**